code/controllers/Processes/timer.dm

```diff
@@ -98,3 +98,11 @@ DECLARE_GLOBAL_CONTROLLER(timer, timer_master)
 			qdel(event)
 			return 1
 	return 0
+
+/proc/getRemainingTime(timer)
+	if(!timer)
+		return FALSE
+	for(var/datum/timedevent/event in timer_master.processing_timers)
```

Comment:
In this case it was a timer being used for tracking how long something is happening on a non obj, this means a process was not an option. The only other option I had was to write a whole new controller, similar to the shuttle controller. The point of this code would be to be able to display in world the remaining time on the timer. Similar to displaying the time remaining on a shuttle.